rust/lsp-lib/src/lsp_plugin.rs

```diff
@@ -311,6 +288,15 @@ impl Plugin for LspPlugin {
 
     fn did_close(&mut self, view: &View<Self::Cache>) {
         eprintln!("close view {}", view.get_id());
+
+        if let Some(view_info) = self.view_info.get(&view.get_id()) {
+            let ls_client = self
```

Comment:
I'm noticing you have to write these ~4 lines _a lot_..

Reply:
I added a helper function that do take rest of the code in a closure to clean this up (see below). Will use that here as well.